twine/commands/check.py

```diff
@@ -76,14 +77,15 @@ def _parse_content_type(value: str) -> Tuple[str, Dict[str, str]]:
 
 def _check_file(
     filename: str, render_warning_stream: _WarningStream
-) -> Tuple[List[str], bool]:
```

Comment:
That we're changing this signature is a sign that we need to do more here. 

Check file previously just cared about the readme and to a lesser extent the metadata version. 

I'd argue we should instead have check readme and check classifiers functions. I'd also argue for a structured data object for representing a finding, the severity and confidence instead of returning lists of strings.

Reply:
As far as I understand, `_check_file` takes the whole distribution file name as argument (e.g. `dist/test-package-0.0.1.tar.gz`), extracts the metadata from it, and run some check on these. AFAIU, it does not seem to read the readme file, the description comes from the metadata as RestructuredText/Markdown, and it tries to render it as HTML.

Wouldn't a function called `_check_readme` be misleading? Can the description be inlined? I need to read more about the spec...

A structured data object representing the result makes sense to me, I'm not a big fan of returning tuples, even less tuples of list/dict. Sounds like a good candidate would be a dataclass or `typing.NamedTuple`, I think either can be used here as Twine currently supports Python 3.8+. Any preference?